datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaConnectorDiagUtils.java

```diff
@@ -61,4 +61,41 @@ public static String reduceTopicPartitionStatsResponses(Map<String, String> resp
 
     return JsonUtils.toJson(result.values());
   }
+
+  /**
+   * Reduce/Merge the KafkaConsumerOffsetsResponse responses of a collection of hosts/instances into one response
+   */
+  public static String reduceConsumerOffsetsResponses(Map<String, String> responses, Logger logger) {
+    Map<String, KafkaConsumerOffsetsResponse> result = new HashMap<>();
+
+    responses.forEach((instance, json) -> {
+      List<KafkaConsumerOffsetsResponse> responseList;
+      try {
+        responseList = JsonUtils.fromJson(json, new TypeReference<List<KafkaConsumerOffsetsResponse>>() {
+        });
+      } catch (Exception e) {
+        logger.error("Invalid response {} from instance {}", json, instance);
+        return;
+      }
+
+      responseList.forEach(response -> {
+        if (response.getConsumerOffsets() == null || StringUtils.isBlank(response.getConsumerGroupId())) {
```

Comment:
Should we ignore all the responses from an instance if one result is bad?

Reply:
I think you're right @vishwajith-s . Will make changes to just log a warning message and not ignore the results from the rest of instances.